scripts/addr2line.py

```diff
@@ -83,41 +84,148 @@ def __call__(self, address):
+        self.sym_addrs : tuple[int]
+        self.sym_names : tuple[str]
+        self.sym_addrs, self.sym_names = zip(*syms) # type: ignore
+
+
+    def __call__(self, addrstr):
+        if self.error:
+            return addrstr + '\n'
+
+        sa = self.sym_addrs
+        sn = self.sym_names
+        slen = len(sa)
+        address = int(addrstr, 16)
+        # print('resolve address', address)
+        idx = bisect.bisect_right(sa, address) - 1
+        assert idx >= -1
+        assert idx < slen
+        if idx == -1:
+            return f'{addrstr} ({sa[0] - address} bytes before first symbol)\n'
+        if idx == slen - 1:
+            # We can easily detect symbol addresses which are too small: they fall before
+            # the first symbol in kallsyms, but for too large it is harder: we can't really
+            # distinguish between an address that is in the *very last* function in the symbol map
+            # and one which is beyond that, since kallsyms doesn't include sybol size. Instead
+            # we use a bit of a quick and dirty heuristic: if the symbol is *far enough* beyond
+            # the last symbol we assume it is not valid. Most likely, the overwhemling majority
+            # of cases are invalid (e.g., due to KASLR) as the final symbol in the map is usually
+            # something obscure.
+            lastsym = sa[-1]
+            if address - lastsym > self.LAST_SYMBOL_MAX_SIZE:
+                return f'{addrstr} ({address - lastsym} bytes after last symbol)\n'
+        saddr = sa[idx]
+        assert saddr <= address
+        return '{}+0x{:x}\n'.format(sn[idx], address - saddr)
+
+
 class BacktraceResolver(object):
 
     class BacktraceParser(object):
         class Type(Enum):
             ADDRESS = 1
             SEPARATOR = 2
+            KERNEL_ADDRESS = 3
 
         def __init__(self):
             addr = "0x[0-9a-f]+"
-            path = "\S+"
-            token = f"(?:{path}\+)?{addr}"
-            full_addr_match = f"(?:({path})\+)?({addr})"
-            self.oneline_re = re.compile(f"^((?:.*(?:(?:at|backtrace):?|:))?(?:\s+))?({token}(?:\s+{token})*)(?:\).*|\s*)$", flags=re.IGNORECASE)
+            path = r"\S+"
+            token = fr"(?:{path}\+)?{addr}"
+            full_addr_match = fr"(?:({path})\+)?({addr})"
+            self.oneline_re = re.compile(
+                fr"^((?:.*(?:(?:at|backtrace):?|:))?(?:\s+))?({token}(?:\s+{token})*)(?:\).*|\s*)$",
+                flags=re.IGNORECASE)
             self.address_re = re.compile(full_addr_match, flags=re.IGNORECASE)
-            self.asan_re = re.compile(f"^(?:.*\s+)\(({full_addr_match})\)\s*$", flags=re.IGNORECASE)
-            self.separator_re = re.compile('^\W*-+\W*$')
+            self.asan_re = re.compile(
+                fr"^(?:.*\s+)\(({full_addr_match})\)\s*$", flags=re.IGNORECASE)
+            self.kernel_re = re.compile(fr'^kernel callstack: (?P<addrs>(?:{addr}\s*)+)$')
+            self.separator_re = re.compile(r'^\W*-+\W*$')
```

Comment:
Except for `kernel_re`, which is new, the other changes here just add `r` ("raw") to the string prefix because these were all escaped wrongly before.